doc/lua_api.txt

```diff
@@ -5671,6 +5678,19 @@ This is basically a reference to a C++ `ServerActiveObject`
 * `hud_set_hotbar_selected_image(texturename)`
     * sets image for selected item of hotbar
 * `hud_get_hotbar_selected_image`: returns texturename
+* `set_minimap_modes({modes}, selected_mode)`: Sets available minimap modes and
+  selected mode number (existing modes are replaced). Modes is a table of mode
+	descriptions. Mode description is a table with four fields :
+    * `type`: Available type:
+      * `off`: Minimap off
+      * `surface`: Minimap in surface mode
+      * `radar`: Minimap in radar mode
+      * `texture`: Texture to be displayed instead of terrain map
+    * `label`: Label associated with this mode (displayed on minimap mode toggle)
```

Comment:
As an alternative to `label` I'd suggest to use two translation strings:
`Minimap in surface mode, Zoom x%g` and
`Minimap in radar mode, Zoom x%g`.

The zoom factor can be calculated automatically and you don't need to worry about translating that stuff as soon the parser script can find those two somewhere in the code. I can see this param would give modders some more freedoms on naming the modes, but why would they need to be changed? The mode types are always the same - only the zoom may change.

Reply:
This label is usefull for naming texture maps. I could make it optional, using your suggestions according to the mode. Problem is, how to determine zoom factor.
Default surface modes are : x1=256 nodes, x2=128 nodes, x4=64 nodes, and its double for radar modes. Imagine a modder wants to have, lets say, 512 nodes and 256 nodes for surface modes. What should be their zoom factor ?

Reply:
I'd say that would be `0.5x` and `1x`. Taking 256 as surface reference.
It is indeed helpful to label texture maps, but not regular surface/radar maps. There it's common to have a string template to fill in the active zoom factor.

Reply:
Ok, I'll take 256 as surface mode reference and 512 as radar mode reference for default labels. It'll be up to modders to do whatever they want if using custom sizes.